### PR TITLE
feature/moved ai models to ai_settigs

### PR DESCRIPTION
**Pull request by gecBurton**
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: redbox-core/redbox/models/chain.py [hunk 1]
```diff
@@ -128,14 +128,22 @@ class AISettings(BaseModel):
...
-    llm_max_tokens: int = 1024
 
-    # size: int = 19 rag_k
     match_boost: int = 1
-    # num_candidates: int = 13 rag_num_candidates
     knn_boost: int = 1
     similarity_threshold: int = 0
 
+    chat_backend: Literal["azure", "openai", "fake"] = "azure"
+    fake_backend_responses: list = ["hello"]
```
Comment (gecBurton) [inline redbox-core/redbox/models/chain.py:137]: I hate me too... but I cant think of another way. all ideas welcome

File: redbox-core/redbox/chains/components.py [hunk 2]
```diff
@@ -1,20 +1,60 @@
...
-        api_key=convert_to_secret_str(env.azure_openai_api_key),
-        azure_endpoint=env.azure_openai_endpoint,
-        model=env.azure_openai_model,
-    )
+from langchain_core.language_models.fake_chat_models import GenericFakeChatModel
+
+log = logging.getLogger()
+
+
+def get_llm(ai_settings: AISettings) -> ChatLiteLLM:
```
Comment (jamesrichards4) [inline redbox-core/redbox/chains/components.py:20]: We shouldn't make the LLM object every time we need one. Adding functools.lru_cache(4) should means we only get a different one if the settings change up to some max (4 here).

Might have to set AISettings to be frozen to be hashable for lru_cache to work but that's what we want anyway

File: redbox-core/redbox/chains/graph.py [hunk 3]
```diff
@@ -100,19 +100,25 @@ def set_prompt_args(state: ChainState):
...
-    )
+    @chain
+    def _build_llm_chain(chain_state: ChainState):
+        llm = get_llm(chain_state["query"].ai_settings)
+        _llm = llm.with_config(tags=["response_flag"]) if final_response_chain else llm
+
+        return RunnableParallel(
+            {
+                "response": make_chat_prompt_from_messages_runnable(
+                    tokeniser=tokeniser, llm_max_tokens=chain_state["query"].ai_settings.llm_max_tokens
```
Comment (jamesrichards4) [inline redbox-core/redbox/chains/graph.py:114]: if max tokens is in the state can't we just get it inside this function and avoid passing it separately?

File: redbox-core/redbox/models/chain.py (hunk 1 above)
Comment (jamesrichards4) [reply redbox-core/redbox/models/chain.py:137]: Could we have the fake llm expect an env var and read it's responses from there? In tests we could set that var for the test/session as needed

**Pull request closed without merging**